add denormalize

drujensen opened the pull request:
This PR adds several new methods to help `normalize` and `denormalize` the input and output.  In order to use a trained model that requires normalization, you need to provide the module normalized inputs.  Also, the output will be normalized, so you need to denormalize it to get a proper prediction:
```
results = model.run(training.normalize_inputs([1.47]))
puts training.denormalize_outputs(results)
```

drujensen commented on src/shainet/basic/data.cr:
```diff
@@ -64,34 +69,67 @@ module SHAInet
+      end
+    end
+
+    def normalize_inputs(inputs : Array(Float64))
```
let me know if you can think of better names for these methods.

bararchy commented:
@drujensen This looks fine, can you add a small spec to check the data gets de-normalized correctly?

bararchy commented:
Also I believe we just broke your PR, can you fix conflicts ?

drujensen commented:
@bararchy I added the tests and cleaned up some of the initialization of the new CNNData that inherits from the Data.  I removed the extraneous initializers and called `super` to call the parent initializer.

bararchy commented:
@ArtLinkov It looks great to me, take a look and let me know if were OK to merge here
